codejail/languages.py

```diff
@@ -0,0 +1,138 @@
+"""
+Language-specific configuration for codejails.
+
+A language consists of:
+
+    `name` A human readable name for the language.
+
+    `argv` A list of extra language-arguments to pass to the Jail's
+    executable.
+
+    `safe_exec_template` Wrapper code to make safe_exec work properly.  If a
+    language has safe_exec_template set to None, commands configured to use
+    that language will not be able to use `codejail.jail.Jail.safe_exec`, and
+    will need to use `codejail.jail.Jail.jail_code` instead.
+
+Preconfigured `Language` objects exist for `python2`, `python3`, and `other`, but
+applications can define their own languages by instantiating `Language`.
+"""
+
+from collections import namedtuple
+import textwrap
+
+
+Language = namedtuple('Language', ['name', 'argv', 'safe_exec_template'])
+
+
+# Allow lowercased names at module level
+# pylint: disable=invalid-name
+
+other = Language(
+    name='other',
+    argv=[],
+    safe_exec_template=None,
+)
+
+python2 = Language(
+    name='python2',
+    argv=[
+        '-E',  # Ignore the environment variables PYTHON*
+        '-B',  # Don't write .pyc files.
+    ],
+    safe_exec_template=textwrap.dedent(
```

Comment:
Yeah. I do see how it's problematic.  I'm not sure of an elegant solution.  We could move the string to an external file, and read it in, but it will still need to be a template we can modify.  It could be a jinja2 or other template instead of a format string, which might provide a bit more flexibility, but the syntax still won't pass a linter.

As a side note, I used % formatting instead of .format() formatting, because .format() doesn't play well with dict literals.